subsys/bluetooth/audio/bap_scan_delegator.c

```diff
@@ -679,24 +755,30 @@ static int scan_delegator_add_source(struct bt_conn *conn,
 
 			LOG_DBG("PA sync %u from %p was rejected with reason %d", pa_sync,
 				(void *)conn, err);
-
-			return BT_GATT_ERR(BT_ATT_ERR_WRITE_REQ_REJECTED);
+			ret = BT_GATT_ERR(BT_ATT_ERR_WRITE_REQ_REJECTED);
+			goto unlock_return;
 		}
 	}
 
 	LOG_DBG("Index %u: New source added: ID 0x%02x",
 		internal_state->index, state->src_id);
 
-	atomic_clear_bit(internal_state->flags,
-			 BASS_RECV_STATE_INTERNAL_FLAG_NOTIFY_PEND);
+	set_receive_state_changed(internal_state);
 
-	receive_state_updated(conn, internal_state);
+unlock_return:
+	err = k_mutex_unlock(&internal_state->mutex);
+	__ASSERT(err == 0, "Failed to unlock mutex: %d", err);
 
-	if (bis_sync_requested) {
-		bis_sync_request_updated(conn, internal_state);
+	if (ret == 0) {
```

Comment:
Since `ret` here refer to a GATT/ATT return values, we could compare it to `BT_ATT_ERR_SUCCESS` instead of `0` to indicate this better

Reply:
This can applied to all places where you've added a `ret`. 

Maybe even with `BT_GATT_ERR(BT_ATT_ERR_SUCCESS)` to be fully comparable, but not a requirement.